File: app/src/main/java/com/nytimes/android/external/register/APIOverridesDelegate.java
package com.nytimes.android.external.register;

import android.support.annotation.IdRes;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
class APIOverridesDelegate {

    private final APIOverrides apiOverrides;

    @Inject
    APIOverridesDelegate(APIOverrides apiOverrides) {
        this.apiOverrides = apiOverrides;
    }

    void setApiOverridesValue(@IdRes int containerLayoutId, ConfigResponse item) {
        switch (containerLayoutId) {
            case R.id.isBillingSupported:
                apiOverrides.setIsBillingSupportedResponse(item.getResponseCode());
                break;
            case R.id.getBuyIntent:
                apiOverrides.setGetBuyIntentResponse(item.getResponseCode());
                break;
            case R.id.buy:
                apiOverrides.setBuyResponse(item.getResponseCode());
                break;
            case R.id.getPurchases:
                apiOverrides.setGetPurchasesResponse(item.getResponseCode());
                break;
            case R.id.getSkuDetails:
                apiOverrides.setGetSkuDetailsResponse(item.getResponseCode());
                break;
            case R.id.consumePurchase:
                apiOverrides.setConsumePurchaseResponse(item.getResponseCode());
                break;
            case R.id.getBuyIntentToReplaceSkus:
                apiOverrides.setGetBuyIntentToReplaceSkusResponse(item.getResponseCode());
                break;
            default:
                // unknown id
                break;
        }
    }

    int getApiOverridesValue(@IdRes int containerLayoutId) {
        int apiValue;
        switch (containerLayoutId) {
            case R.id.isBillingSupported:
                apiValue = apiOverrides.getIsBillingSupportedResponse();
                break;
            case R.id.getBuyIntent:
                apiValue = apiOverrides.getGetBuyIntentResponse();
                break;
            case R.id.buy:
                apiValue = apiOverrides.getBuyResponse();
                break;
            case R.id.getPurchases:
                apiValue = apiOverrides.getGetPurchasesResponse();
                break;
            case R.id.getSkuDetails:
                apiValue = apiOverrides.getGetSkuDetailsResponse();
                break;
            case R.id.consumePurchase:
                apiValue = apiOverrides.getConsumePurchaseResponse();
                break;
            case R.id.getBuyIntentToReplaceSkus:
                apiValue = apiOverrides.getGetBuyIntentToReplaceSkusResponse();
                break;
            default:
                apiValue = -1;
                break;
        }
        return apiValue;
    }
}

File: app/src/main/java/com/nytimes/android/external/register/APIOverridesDelegate.kt
package com.nytimes.android.external.register

import androidx.annotation.IdRes

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class APIOverridesDelegate @Inject constructor(private val apiOverrides: APIOverrides) {

    fun setApiOverridesValue(@IdRes containerLayoutId: Int, item: ConfigResponse) {
        when (containerLayoutId) {
            R.id.nyt_register_isBillingSupported -> apiOverrides.isBillingSupportedResponse = item.responseCode
            R.id.nyt_register_getBuyIntent -> apiOverrides.getBuyIntentResponse = item.responseCode
            R.id.nyt_register_buy -> apiOverrides.buyResponse = item.responseCode
            R.id.nyt_register_getPurchases -> apiOverrides.getPurchasesResponse = item.responseCode
            R.id.nyt_register_getSkuDetails -> apiOverrides.getSkuDetailsResponse = item.responseCode
            R.id.nyt_register_consumePurchase -> apiOverrides.consumePurchaseResponse = item.responseCode
            R.id.nyt_register_getBuyIntentToReplaceSkus -> apiOverrides.getBuyIntentToReplaceSkusResponse = item.responseCode
            else -> {
                // No Op - Unknown id
            }
        }
    }

    fun getApiOverridesValue(@IdRes containerLayoutId: Int): Int {
        return when (containerLayoutId) {
            R.id.nyt_register_isBillingSupported -> apiOverrides.isBillingSupportedResponse
            R.id.nyt_register_getBuyIntent -> apiOverrides.getBuyIntentResponse
            R.id.nyt_register_buy -> apiOverrides.buyResponse
            R.id.nyt_register_getPurchases -> apiOverrides.getPurchasesResponse
            R.id.nyt_register_getSkuDetails -> apiOverrides.getSkuDetailsResponse
            R.id.nyt_register_consumePurchase -> apiOverrides.consumePurchaseResponse
            R.id.nyt_register_getBuyIntentToReplaceSkus -> apiOverrides.getBuyIntentToReplaceSkusResponse
            else -> -1
        }
    }
}
